geometric/optimize.py

```diff
@@ -1881,7 +1874,14 @@ def main():
     args = parser.parse_args(sys.argv[1:])
 
     # Run the optimizer.
-    run_optimizer(**vars(args))
+    try:
+        run_optimizer(**vars(args))
+    except EngineError:
+        logger.info("EngineError:\n" + traceback.format_exc())
+        sys.exit(51)
+    except GeomOptNotConvergedError:
+        logger.info("Geometry Converge Failed Error:\n" + traceback.format_exc())
+        sys.exit(50)
```

Comment:
You might want a final general `except` for unknown cases.

Reply:
Sure. Updated